toolkit/Makefile

```
@@ -19,16 +21,29 @@ PACKAGE_REBUILD_LIST    ?=
 PACKAGE_IGNORE_LIST     ?=
 SRPM_PACK_LIST          ?=
 
-REBUILD_TOOLCHAIN               ?= n
+# Logic to auto configure build options for optimized builds
+# These will set the default values for: REBUILD_TOOLS, REBUILD_TOOLCHAIN, DELTA_BUILD, INCREMENTAL_TOOLCHAIN, and ALLOW_TOOLCHAIN_DOWNLOAD_FAIL
+include ./quickrebuild_autoconfig.mk
+
+# We generally want to use the DELTA_BUILD flag to determine whether to use the incremental toolchain, but to preserve
+# backwards compatibility, we'll still use the INCREMENTAL_TOOLCHAIN flag if it's set.
+DELTA_BUILD                     ?= n
+ifeq ($(origin DELTA_BUILD), undefined)
+# We didn't set DELTA_BUILD, so use the previous default behavior of INCREMENTAL_TOOLCHAIN for backwards compatibility.
 INCREMENTAL_TOOLCHAIN           ?= n
+else
+# We set DELTA_BUILD, so inherit value for INCREMENTAL_TOOLCHAIN.
+INCREMENTAL_TOOLCHAIN           ?= $(DELTA_BUILD)
+endif
```

I think I'm missing a crucial piece of `make` magic. With `DELTA_BUILD ?= n` at the very top, in my local tests the `ifeq ($(origin DELTA_BUILD), undefined)` block was never executed. It was fine, if I moved the default `DELTA_BUILD` to somewhere after that check.

And what would fail with this approach:
```suggestion
DELTA_BUILD                     ?= n
INCREMENTAL_TOOLCHAIN           ?= $(DELTA_BUILD)
```

I think this still allows the user to set `INCREMENTAL_TOOLCHAIN` independently from `DELTA_BUILD` with the default being `n` for both.

---

Btw. using `ifneq ($(origin DELTA_BUILD), command line)` seems to made `make` go into that block, if I didn't set `DELTA_BUILD` from the command line.

---

Cant see a reason your simple solution doesn't work, will use that.